dart/lib/src/protocol/message.dart

```diff
@@ -20,10 +20,20 @@ class Message {
   Message(this.formatted, {this.template, this.params});
 
   Map<String, dynamic> toJson() {
-    return {
-      'formatted': formatted,
-      'message': template,
-      'params': params,
-    };
+    final json = <String, dynamic>{};
+
+    if (formatted != null && formatted.isNotEmpty) {
+      json['formatted'] = formatted;
+    }
+
+    if (template != null && template.isNotEmpty) {
+      json['message'] = template;
+    }
```

Comment:
we don't need to `isNotEmpty` for `Strings` but only for collections

Reply:
```suggestion
    if (formatted != null) {
      json['formatted'] = formatted;
    }

    if (template != null) {
      json['message'] = template;
    }
```